### ISSUE TITLE
Newline characters not parsed by lineinfile module

### ISSUE THREAD

**Issue by billmcmath** [ansible/ansible#8289]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.6.9 (release1.6.9 663ecd0d04) last updated 2014/07/25 11:56:58 (GMT +100)
##### Environment:

Mac OS X 10.9.4 managing CentOS 6.5 (PVHVM)
##### Summary:

When a newline character `\n` is included in the line parameter text of the lineinfile module, it is being written as `\n` into the file rather than as a new line character.
##### Steps To Reproduce:

Add the following to the play

```
lineinfile: dest=/etc/ssh/sshd_config state=present
regexp="^Match User username$"
line="Match User username\n\tPasswordAuthentication yes"
insertafter=EOF
```
##### Expected Results:

```
Match User username
    PasswordAuthentication yes
```

at the end of the file
##### Actual Results:

```
Match User username\n\tPasswordAuthentication yes
```

at the end of the file.

**Comment by jimi-c** [comment]
Hrm, using the following sample (based mainly on yours), this works as expected for me:

<pre>
- hosts: localhost
  connection: local
  gather_facts: no
  tasks:
  - copy: dest=/tmp/8289.txt content="Match User username\n"
  - lineinfile: dest=/tmp/8289.txt state=present regexp="^Match User username$" line="Match User username\n\tPasswordAuthentication yes" insertafter=EOF
$ ansible-playbook -vv 8289.yml 
PLAY [localhost] ************************************************************** 
TASK: [copy dest=/tmp/8289.txt content="Match User username\n"] *************** 
changed: [127.0.0.1] => {"changed": true, "dest": "/tmp/8289.txt", "gid": 0, "group": "root", "md5sum": "5b65ac446fbfac21915c94747c8fb7fb", "mode": "0644", "owner": "root", "secontext": "unconfined_u:object_r:user_tmp_t:s0", "size": 20, "src": "/root/.ansible/tmp/ansible-tmp-1406300229.48-213254305810812/source", "state": "file", "uid": 0}
TASK: [lineinfile dest=/tmp/8289.txt state=present regexp="^Match User username$" line="Match User username\n\tPasswordAuthentication yes" insertafter=EOF] *** 
<127.0.0.1> REMOTE_MODULE lineinfile dest=/tmp/8289.txt state=present regexp="^Match User username$" line="Match User username\n\tPasswordAuthentication yes" insertafter=EOF
changed: [127.0.0.1] => {"backup": "", "changed": true, "msg": "line replaced"}
PLAY RECAP ******************************************************************** 
127.0.0.1                  : ok=2    changed=2    unreachable=0    failed=0   
$ cat /tmp/8289.txt 
Match User username
    PasswordAuthentication yes
</pre>

I am running on a linux host though, but we also have an integration test that specifically uses newlines with lineinfile that is passing across multiple other linux distribution flavors. I'm doubtful this would be an OSX specific issue, but can you try on another OS to see if it still fails?

**Comment by billmcmath** [comment]
Hi @jimi-c 

I've just run this on a CentOS 6.5 VM:

```
[root@cloud-server-02 ansible]# ansible-playbook -vv 8289.yml
ERROR: Unable to find an inventory file, specify one with -i ?
[root@cloud-server-02 ansible]# ansible-playbook -vv 8289.yml -i hosts 

PLAY [localhost] ************************************************************** 

TASK: [copy dest=/tmp/8289.txt content="Match User username\n"] *************** 
changed: [localhost] => {"changed": true, "dest": "/tmp/8289.txt", "gid": 0, "group": "root", "md5sum":     "5b65ac446fbfac21915c94747c8fb7fb", "mode": "0644", "owner": "root", "size": 20, "src":     "/root/.ansible/tmp/ansible-tmp-1406486695.78-20224841642136/source", "state": "file", "uid": 0}

TASK: [lineinfile dest=/tmp/8289.txt state=present regexp="^Match User username$" line="Match     User username\n\tPasswordAuthentication yes" insertafter=EOF] *** 
<localhost> REMOTE_MODULE lineinfile dest=/tmp/8289.txt state=present regexp="^Match User     username$" line="Match User username\n\tPasswordAuthentication yes" insertafter=EOF
changed: [localhost] => {"backup": "", "changed": true, "msg": "line replaced"}

PLAY RECAP ******************************************************************** 
localhost                  : ok=2    changed=2    unreachable=0    failed=0   

[root@cloud-server-02 ansible]# cat /tmp/8289.txt 
Match User username\n\tPasswordAuthentication yes
[root@cloud-server-02 ansible]# ansible --version
ansible 1.6.9 (release1.6.9 663ecd0d04) last updated 2014/07/27 18:37:29 (GMT +000)
[root@cloud-server-02 ansible]# python --version
Python 2.6.6
[root@cloud-server-02 ansible]# uname -a
Linux cloud-server-02 2.6.32-431.20.3.el6.x86_64 #1 SMP Thu Jun 19 21:14:45 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by billmcmath** [comment]
Hi @jimi-c ,

I just did a little test, I only just noticed that version `release1.6.10` is available so I switched to it, I was seeing the same behaviour in it too, I then moved to devel to see if the problem was there too.  Devel is fine, the file at the end had the newline character and not `\n`.

Well I have a way past this now, use devel until `release1.6.11` or `release1.7` are out.

Thanks,
Bill
